### ISSUE TITLE
CPU runs fail with explicit --devices due to list parsing and fixed DDP strategy in CLI

### ISSUE THREAD

**Issue by kshirajahere** [mllam/neural-lam#393]
### Summary
`neural_lam/train_model.py` currently parses `--devices` as a list and passes it directly to Lightning, while also hardcoding `strategy="ddp"`.

On CPU this is invalid in common local setups:
- Lightning expects an integer device count for CPU (not a list like `[1]`).
- DDP is not appropriate as a default for single-device CPU runs.

This causes CPU workflows (including local debugging/CI-style runs) to fail before training starts.

### Why this is urgent
- Blocks CPU-only users from running train/eval via CLI with explicit `--devices`.
- Creates avoidable launch failures in non-GPU environments.
- Prevents simple local reproducibility checks when no CUDA device is available.

### Reproduction
From repo root, run (example):
```bash
python -m neural_lam.train_model --dataset dummy --devices 1 --num_epochs 1
```
(or equivalent CLI invocation using CPU)

### Current behavior
Trainer initialization fails due to invalid CPU `devices` type and/or strategy mismatch.

### Expected behavior
- CPU path should accept `--devices` as a single positive integer count.
- Strategy should be selected contextually:
  - `auto` for CPU and single-device runs
  - `ddp` for multi-device CUDA runs

### Related but not duplicate
There is an existing issue around hardcoded multi-GPU assumptions in tests (#192), but this report targets runtime CLI behavior on CPU with explicit `--devices` parsing + strategy selection.

### Proposed fix
Add centralized device/strategy resolution in CLI before Trainer construction:
- Normalize CPU devices to integer
- Validate CPU does not receive multiple explicit device IDs
- Choose `strategy` based on accelerator/device count

**Comment by Shyam-Sunder-saini** [comment]
Hi, I looked into the problem described in #393 around the handling of `--devices` and the hardcoded `ddp` strategy in `train_model.py`.

What was happening:-

The CLI parses `--devices` as a list and passes it directly to Lightning while the trainer uses `strategy="ddp"` by default.
On CPU environments this can lead to failures because:

* Lightning expects an **integer device count for CPU**, not a list.
* `ddp` is not appropriate for **single-device CPU runs**.
* As a result, CPU-only workflows (local debugging / CI-style runs) may fail before training starts.

What I changed:

* Normalized the parsed `--devices` argument:

  * `"auto"` remains `"auto"`
  * explicit device IDs are converted to integers.
* Added logic to resolve the **accelerator automatically** (`cpu` / `gpu`).
* Selected the training **strategy dynamically**:

  * `auto` for CPU or single-device runs
  * `ddp` only when running multi-device CUDA setups.

Result:

* CPU-only runs now start correctly via the CLI.
* Multi-GPU training behavior remains unchanged.
* The existing `--devices` CLI interface is preserved.

Local test:

Tested locally on CPU with:
```
python -m neural_lam.train_model --config_path tests/datastore_examples/mdp/danra_100m_winds/config.yaml --devices auto --epochs 1  
```
Training starts successfully and the trainer initializes without device/strategy errors.

Let me know if this approach aligns with the intended fix for this issue.
